app/vmalert/config/config_test.go

```diff
@@ -15,13 +15,13 @@ import (
 )
 
 func TestMain(m *testing.M) {
-	u, _ := url.Parse("https://victoriametrics.com/path")
-	notifier.InitTemplateFunc(u)
+	tmpl, _ := notifier.LoadTemplates([]string{"testdata/*good.tmpl"})
+	notifier.SetTemplate(tmpl)
```

Comment:
Why would you separate `Load` and `Set` methods?

Reply:
I think, it would be worth to try to move out all this templating stuff to the separate package `templates` or something like this.
This package could hold the loaded templates configuration and its checksum, in the same way as `notifier` package does for notifier config. So on config reload time there will be only one method `templates.Reload()` or something like this.
This package will also have its own tests.
I think, it could reduce cognitive load while reading `notifier` package, since it already contains too much things.